src/errors/ParseError.ts

```diff
@@ -0,0 +1,17 @@
+type ConstructorParams<ResponseBodyType = unknown> = {
+  message: string;
+  responseBody: ResponseBodyType;
+}
+
+class ParseError<ResponseBodyType = unknown> extends Error {
```

Comment:
What is the idea behind generic body type? Isn't ParseError thrown only when we try to parse incorrectly formatted json?

Reply:
Not quite. The library will throw in other cases as well. As long as option to parse body is enabled, such error could be thrown by the library. Here is the reference code - https://github.com/sindresorhus/got/blob/efafa167d70bb68591cba89343d3fae9a69d8cb1/source/core/response.ts#L129

Reply:
Well, we use only `text` and `json`. In the case of of `text` the code can't throw because rawBody is a buffer, so doesn't this leave only json parsing error?

Reply:
This could be boiled down just to `unknown`, since got's `Response` type is generic and this is only well fit hard cast.